addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1955,6 +1984,8 @@ const SnippetOptionWidget = Widget.extend({
         if (widgetValue) {
             this.$target[0].classList.add(...widgetValue.trim().split(/\s+/g));
         }
+
+        if (previewMode === false) await this.updateChangesInWysiwyg();
```

Comment:
@Goaman and @dmo-odoo can clarify this better than I could. This basically applies changes to the editor's abstraction.
As for the pointers, we will be working on a comprehensive dev doc as soon as possible.

Reply:
When the changes of the option are validated (`previewMode === false`) we call `updateChangesInWysiwyg` that:
1) update all the attributes of `this.$target` in the dom to our VirtualDocument
2) discard everything inside this.$target in our VDoc
3) send all the innerHTML of this.$target in our VDoc

The reason is to allow the options code to be unchanged and while update our VDoc.
However, there is big problem with this strategy as some options will have their targed being invalidated (because of step 2).

We are working to make the API easier to use my implementing a mutationObserver of the dom so you can call something like:

```
editor.observeChange(()=> {
  element.addClass(...)
});
```

Which will be translated directly in our VDoc and will not change any dom reference that an option could have. Which will remove logic required for the strategy above.

More on that topic when the mutationObserver strategy will be implemented.